src/calendar/index.ts

```diff
@@ -210,20 +218,58 @@ export class CalendarBase<P extends CalendarProperties = CalendarProperties> ext
 			case Keys.Enter:
 			case Keys.Space:
 				const { onDateSelect } = this.properties;
-				onDateSelect && onDateSelect(new Date(year, month, this._focusedDay));
+				const date = new Date(year, month, this._focusedDay);
+				if (this._dateInMinMax(date)) {
+					onDateSelect && onDateSelect(date);
+				}
 		}
 	}
 
+	private _lastMonthInMin() {
+		let {
+			year,
+			month
+		} = this._getMonthYear();
+		let { minDate } = this.properties;
+
+		return monthInMin(year, month - 1, minDate);
+	}
+
+	private _nextMonthInMax() {
+		let {
+			year,
+			month
+		} = this._getMonthYear();
+		const { maxDate } = this.properties;
+
+		return monthInMax(year, month + 1, maxDate);
+	}
+
+	private _dateInMinMax(date: Date): boolean {
+		const {
+			minDate = date,
+			maxDate = date
+		} = this.properties;
+
+		const inRange = date >= minDate && date <= maxDate;
+		return inRange;
+	}
+
 	private _onMonthDecrement() {
 		const {
 			month,
 			year
 		} = this._getMonthYear();
 		const {
 			onMonthChange,
-			onYearChange
+			onYearChange,
+			minDate
 		} = this.properties;
 
+		if (!monthInMin(year, month - 1, minDate)) {
```

Comment:
I don't think that it is possible to reach inside of this block (I wasn't able to figure out how to test it). We may be able to remove it.

Same with line [295](https://github.com/dojo/widgets/pull/620/files#diff-663a5c4d6e355c4bd09383f0b5119996R295).